server/src/main/java/org/elasticsearch/index/engine/ReadOnlyEngine.java

```
@@ -0,0 +1,349 @@
+        final SequenceNumbers.CommitInfo seqNoStats =
+            SequenceNumbers.loadSeqNoInfoFromLuceneCommit(infos.userData.entrySet());
+        long maxSeqNo = seqNoStats.maxSeqNo;
+        long localCheckpoint = seqNoStats.localCheckpoint;
+        return new SeqNoStats(maxSeqNo, localCheckpoint, localCheckpoint);
+    }
+
+    @Override
+    public GetResult get(Get get, BiFunction<String, SearcherScope, Searcher> searcherFactory) throws EngineException {
+        return getFromSearcher(get, searcherFactory, SearcherScope.EXTERNAL);
+    }
+
+    @Override
+    protected ReferenceManager<IndexSearcher> getReferenceManager(SearcherScope scope) {
+        return searcherManager;
+    }
+
+    @Override
+    protected SegmentInfos getLastCommittedSegmentInfos() {
+        return lastCommittedSegmentInfos;
+    }
+
+    @Override
+    public String getHistoryUUID() {
+        return lastCommittedSegmentInfos.userData.get(Engine.HISTORY_UUID_KEY);
+    }
+
+    @Override
+    public long getWritingBytes() {
+        return 0;
+    }
+
+    @Override
+    public long getIndexThrottleTimeInMillis() {
+        return 0;
+    }
+
+    @Override
+    public boolean isThrottled() {
+        return false;
+    }
+
+    @Override
+    public IndexResult index(Index index) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public DeleteResult delete(Delete delete) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public NoOpResult noOp(NoOp noOp) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public boolean isTranslogSyncNeeded() {
+        return false;
+    }
+
+    @Override
+    public boolean ensureTranslogSynced(Stream<Translog.Location> locations) {
+        return false;
+    }
+
+    @Override
+    public void syncTranslog() {
+    }
+
+    @Override
+    public Closeable acquireRetentionLockForPeerRecovery() {
+        return () -> {};
```

I think this should be an `assert false;` + throw new UnsupportedOperationException

---

it's totally fine to do a peerRecovery from this  engine.